### PR TITLE
feat(gnovm): returns error like in Golang when assignment with a function which does not return any value

### PR DESCRIPTION
**Pull request by hthieu1110**
This PR aims at resolving this issue: https://github.com/gnolang/gno/issues/1082

This depends on https://github.com/gnolang/gno/pull/3017 because it refactored the code to sync the logic/code between AssignStmt vs ValueDecl.

Related https://github.com/gnolang/gno/pull/2695

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
</details>

### CODE REVIEW

**Comment by hthieu1110** [comment]
Depends on (3017)[https://github.com/gnolang/gno/pull/3017]

**Comment by hthieu1110** [comment]
Related [2695](https://github.com/gnolang/gno/pull/2695)

File: gnovm/pkg/gnolang/type_check.go [hunk 1]
```diff
@@ -1029,6 +1029,14 @@ func assertValidAssignRhs(store Store, last BlockNode, n Node) {
 			tt = evalStaticType(store, last, exp)
 			panic(fmt.Sprintf("%s (type) is not an expression", tt.String()))
 		}
+
+		// Ensures that function used in ValueDecl or AssignStmt must return at least 1 value.
+		if cx, ok := exp.(*CallExpr); ok {
+			tType, ok := tt.(*tupleType)
+			if ok && len(tType.Elts) == 0 {
```
Comment (ltzmaxwell) [inline gnovm/pkg/gnolang/type_check.go:1036]: do you think the same logic in `evalStaticTypeOf` is still needed?

Comment (hthieu1110) [reply gnovm/pkg/gnolang/type_check.go:1036]: IMHO, I think the the login in evalStaticTypeOf and the check in this PR do not target the same check so they should exist both

Comment (hthieu1110) [reply gnovm/pkg/gnolang/type_check.go:1036]: Ah, maybe I misunderstood your comment, did you mean that should we have the same check in `evalStaticTypeOf ` ?

Comment (ltzmaxwell) [reply gnovm/pkg/gnolang/type_check.go:1036]: what's the logic in `evalStaticTypeOf` targeting? I mean if it's not needed we can just remove it, please check, thank you.

Comment (hthieu1110) [reply gnovm/pkg/gnolang/type_check.go:1036]: I've checked, it should be fine to remove the check in `evalStaticTypeOf`. Thanks

**Review by ltzmaxwell — APPROVED** [review]

**Pull request merged**